src/ORM/DataList.php

```diff
@@ -525,14 +526,27 @@ public function filterAny()
             throw new InvalidArgumentException('Incorrect number of arguments passed to filterAny()');
         }
 
-        return $this->alterDataQuery(function (DataQuery $query) use ($whereArguments) {
-            $subquery = $query->disjunctiveGroup();
-
+        $list = $this->alterDataQuery(function (DataQuery $query) use ($whereArguments) {
+            $subquery = $this->getFilterAnySubquery($query, $whereArguments);
             foreach ($whereArguments as $field => $value) {
                 $filter = $this->createSearchFilter($field, $value);
                 $filter->apply($subquery);
             }
         });
+
+        return $list;
+    }
+
+    private function getFilterAnySubquery(DataQuery $query, array $whereArguments): DataQuery_SubGroup
+    {
+        $clause = 'WHERE';
+        foreach (array_keys($whereArguments) as $field) {
+            if (preg_match('#\.(COUNT|SUM|AVG|MIN|MAX)\(#', strtoupper($field))) {
```

Comment:
I won't hold off merging for this, but it probably makes sense to use case insensitive regex instead.
```suggestion
            if (preg_match('#\.(COUNT|SUM|AVG|MIN|MAX)\(#i', $field)) {
```

Reply:
Will keep as is, I think it's easier to read with the strtoupper(), probably milliseconds faster performance as well